tests/src/webhooks/providers/bit2me/index.test.ts

```diff
@@ -0,0 +1,182 @@
+import { FastifyInstance } from 'fastify';
+
+import { closeServerAndDatabase, initializeServerAndDatabase } from '../../../utils/initializeServer';
+import { getCryptoInvoiceWebhook, getCustomer, getInvoice } from '../../../fixtures';
+import jwt from 'jsonwebtoken';
+import config from '../../../../../src/config';
+import { PaymentService } from '../../../../../src/services/payment.service';
+import Stripe from 'stripe';
+import { InvoiceCompletedHandler } from '../../../../../src/webhooks/events/invoices/InvoiceCompletedHandler';
+
+let app: FastifyInstance;
+
+beforeAll(async () => {
+  app = await initializeServerAndDatabase();
+});
+
+afterAll(async () => {
+  await closeServerAndDatabase();
+});
+
+describe('Handling webhook for crypto payments', () => {
+  test('When the foreignId does not match with the invoiceId we get from the encrypted token, then an error indicating so is thrown', async () => {
```

Comment:
Is not encrypted, it's encoded, in any case you get the id from the token, the decoded one